src/Avalonia.Diagnostics/Diagnostics/ViewModels/PropertyViewModel.cs

```diff
@@ -25,19 +26,37 @@ protected static string ConvertToString(object value)
                 return "(null)";
             }
 
-            var converter = TypeDescriptor.GetConverter(value);
-            return converter?.ConvertToString(value) ?? value.ToString();
+            //Check if there's an user provided ToString(), prefer that over the TypeDescriptor conversion
+            if (value.GetType().GetMethod(nameof(ToString), System.Type.EmptyTypes)
```

Comment:
This kinda conflicts with zero reflection long-time goal

Reply:
Should this apply to DevTools though? It kinda relies on reflection to do type introspection, we even have console there.

We should definitely avoid using reflection in core framework but tooling might need to be exempt from this. Opinions?

Reply:
I'm still planning to make devtools useful when app is running on iOS device. So there will be a no-JIT requirement at some point.

Reply:
I wonder if we could have such features pluggable, so we can have fully featured DevTools on desktop and a stripped version on platforms that don't support JIT/Reflection/Etc.

Reply:
How do you want to run DevTools on mobile devices?

Reply:
The same way our previewer currently works on Rider - by listening on a TCP socket and providing a small web page that loads frames and sends input via websocket. The infrastructure is already there.

Reply:
The reflection is gone :)